didn't run successfully on the master branch, here is the output:
```
No config file exists, using default configurations for logger
2019-07-05 22:16:31.597 +02:00 DEBUG sync/once.go:44 init logger...
2019-07-05 22:16:31.600 +02:00 DEBUG go-archaius/configurationfactory.go:90 ConfigurationFactory Initiated
2019-07-05 22:16:31.600 +02:00 INFO go-archaius/archaius.go:53 Configuration files: 
2019-07-05 22:16:31.600 +02:00 WARN config-manager/configurationmanager.go:140 empty configurtion from FileSource
2019-07-05 22:16:31.600 +02:00 INFO config-manager/configurationmanager.go:117 invoke dynamic handler:FileSource
2019-07-05 22:16:31.601 +02:00 INFO go-archaius/archaius.go:112 archaius init success
2019-07-05 22:16:31.601 +02:00 WARN config-manager/configurationmanager.go:140 empty configurtion from MemorySource
2019-07-05 22:16:31.601 +02:00 INFO config-manager/configurationmanager.go:117 invoke dynamic handler:MemorySource
2019-07-05 22:16:31.601 +02:00 INFO file-source/filesource.go:353 create new watcher
2019-07-05 22:16:31.602 +02:00 WARN config-manager/configurationmanager.go:140 empty configurtion from CommandlineSource
2019-07-05 22:16:31.602 +02:00 INFO config-manager/configurationmanager.go:117 invoke dynamic handler:CommandlineSource
2019-07-05 22:16:31.602 +02:00 INFO memory-source/memorysource.go:105 mem source callback prepared
2019-07-05 22:16:31.603 +02:00 INFO config-manager/configurationmanager.go:117 invoke dynamic handler:EnvironmentSource
2019-07-05 22:16:31.603 +02:00 ERROR config/config.go:64 filepath.Walk() returned lstat /tmp/go-build204483611/b001/conf: no such file or directory

=== RUN   TestInitializeConfig
==================
WARNING: DATA RACE
Write at 0x00c000078488 by goroutine 13:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).handlePriority()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:271 +0x2fd
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).handleFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:233 +0x398
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).AddFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:145 +0x493
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius.AddFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/archaius.go:281 +0x138
  github.com/kubeedge/beehive/pkg/common/config_test.addSources.func1()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:86 +0xc7
  path/filepath.walk()
      /usr/local/go/src/path/filepath/path.go:358 +0x501
  path/filepath.walk()
      /usr/local/go/src/path/filepath/path.go:382 +0x39d
  path/filepath.Walk()
      /usr/local/go/src/path/filepath/path.go:404 +0x143
  github.com/kubeedge/beehive/pkg/common/config_test.addSources()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:77 +0x52
  github.com/kubeedge/beehive/pkg/common/config_test.TestInitializeConfig()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:26 +0x174
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163

Previous read at 0x00c000078488 by goroutine 8:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*watch).startWatchPool()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:359 +0x9e

Goroutine 13 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1072 +0x2eb
  main.main()
      _testmain.go:44 +0x222

Goroutine 8 (finished) created at:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).DynamicConfigHandler()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:336 +0xda
==================
==================
WARNING: DATA RACE
Read at 0x00c0000784a8 by goroutine 13:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).handleFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:239 +0x44d
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).AddFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:145 +0x493
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius.AddFile()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/archaius.go:281 +0x138
  github.com/kubeedge/beehive/pkg/common/config_test.addSources.func1()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:86 +0xc7
  path/filepath.walk()
      /usr/local/go/src/path/filepath/path.go:358 +0x501
  path/filepath.walk()
      /usr/local/go/src/path/filepath/path.go:382 +0x39d
  path/filepath.Walk()
      /usr/local/go/src/path/filepath/path.go:404 +0x143
  github.com/kubeedge/beehive/pkg/common/config_test.addSources()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:77 +0x52
  github.com/kubeedge/beehive/pkg/common/config_test.TestInitializeConfig()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config_test.go:26 +0x174
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163

Previous write at 0x00c0000784a8 by goroutine 6:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source.(*fileSource).DynamicConfigHandler()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/sources/file-source/filesource.go:334 +0x8f

Goroutine 13 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1072 +0x2eb
  main.main()
      _testmain.go:44 +0x222

Goroutine 6 (finished) created at:
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/core/config-manager.(*ConfigurationManager).AddSource()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/core/config-manager/configurationmanager.go:118 +0x6ff
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius.(*ConfigFactory).AddSource()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/configurationfactory.go:165 +0xc4
  github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius.Init()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/vendor/github.com/go-chassis/go-archaius/archaius.go:87 +0x2e6
  github.com/kubeedge/beehive/pkg/common/config.InitializeConfig.func1()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config.go:35 +0x54
  sync.(*Once).Do()
      /usr/local/go/src/sync/once.go:44 +0xde
  github.com/kubeedge/beehive/pkg/common/config.InitializeConfig()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config.go:34 +0x46
  github.com/kubeedge/beehive/pkg/common/config.init.0()
      /home/subpath/data/go/src/github.com/kubeedge/beehive/pkg/common/config/config.go:73 +0x49
  github.com/kubeedge/beehive/pkg/common/config.init()
      <autogenerated>:1 +0xd3
  main.init()
      <autogenerated>:1 +0xa1
==================
2019-07-05 22:16:31.605 +02:00 DEBUG config-manager/configurationmanager.go:391 EventReceived &{FileSource CREATE loggerLevel DEBUG}
2019-07-05 22:16:31.605 +02:00 DEBUG event-system/eventsystem.go:115 event generated for a*
2019-07-05 22:16:31.606 +02:00 DEBUG config-manager/configurationmanager.go:391 EventReceived &{FileSource CREATE enableRsyslog %!s(bool=false)}
2019-07-05 22:16:31.606 +02:00 DEBUG event-system/eventsystem.go:115 event generated for a*
2019-07-05 22:16:31.606 +02:00 DEBUG go-archaius/archaius.go:186 config value after change loggerLevel | DEBUG
2019-07-05 22:16:31.606 +02:00 DEBUG config-manager/configurationmanager.go:391 EventReceived &{FileSource CREATE logFormatText %!s(bool=true)}
2019-07-05 22:16:31.607 +02:00 DEBUG event-system/eventsystem.go:115 event generated for a*
2019-07-05 22:16:31.606 +02:00 DEBUG go-archaius/archaius.go:186 config value after change enableRsyslog | %!s(bool=false)
2019-07-05 22:16:31.607 +02:00 DEBUG go-archaius/archaius.go:186 config value after change logFormatText | %!s(bool=true)
2019-07-05 22:16:31.607 +02:00 DEBUG config-manager/configurationmanager.go:391 EventReceived &{FileSource CREATE writers [stdout]}
2019-07-05 22:16:31.608 +02:00 DEBUG event-system/eventsystem.go:115 event generated for a*
2019-07-05 22:16:31.609 +02:00 DEBUG config-manager/configurationmanager.go:391 EventReceived &{FileSource CREATE modules.enabled [eventbus servicebus]}
2019-07-05 22:16:31.609 +02:00 DEBUG event-system/eventsystem.go:115 event generated for a*
2019-07-05 22:16:31.610 +02:00 DEBUG file-source/filesource.go:395 file event /tmp/go-build204483611/b001/conf/modules.yaml, operation is 4. reload it.
--- FAIL: TestInitializeConfig (0.01s)
    testing.go:809: race detected during execution of test
2019-07-05 22:16:31.610 +02:00 WARN file-source/filesource.go:398 the file change mode: "/tmp/go-build204483611/b001/conf/modules.yaml": REMOVE, continue
2019-07-05 22:16:31.610 +02:00 DEBUG file-source/filesource.go:395 file event /tmp/go-build204483611/b001/conf/logging.yaml, operation is 4. reload it.
2019-07-05 22:16:31.610 +02:00 WARN file-source/filesource.go:398 the file change mode: "/tmp/go-build204483611/b001/conf/logging.yaml": REMOVE, continue
FAIL
FAIL	github.com/kubeedge/beehive/pkg/common/config	0.040s
No config file exists, using default configurations for logger
2019-07-05 22:16:30.543 +02:00 DEBUG sync/once.go:44 init logger...
=== RUN   TestLoggerInitilization
2019-07-05 22:16:30.544 +02:00 DEBUG log/logger_test.go:10 testing log
2019-07-05 22:16:30.544 +02:00 DEBUG log/logger_test.go:11 testing beehive
--- PASS: TestLoggerInitilization (0.00s)
PASS
ok  	github.com/kubeedge/beehive/pkg/common/log	1.032s
=== RUN   TestServer
Listening on /tmp/us.socket
--- PASS: TestServer (10.00s)
=== RUN   TestClient
Connected from &{{0xc0000ae200}}zhangqi test
===============response from server: 2019-07-05 22:16:50.565910251 +0200 CEST m=+20.007179219
--- PASS: TestClient (10.00s)
PASS
ok  	github.com/kubeedge/beehive/pkg/common/util	21.045s
?   	github.com/kubeedge/beehive/pkg/core	[no test files]
No config file exists, using default configurations for logger
2019-07-05 22:16:31.458 +02:00 DEBUG sync/once.go:44 init logger...
=== RUN   TestSendSync
2019-07-05 22:16:31.459 +02:00 WARN context/context_channel.go:296 failed to get channel, type(test_dest)
2019-07-05 22:16:31.459 +02:00 WARN context/context_channel.go:81 failed to get channel for module(test_dest)
2019-07-05 22:16:31.459 +02:00 WARN context/context_factory.go:65 failed to receive message
receive msg: {{  0 false} {   } <nil>}, error: failed to get channel for module(test_dest)
2019-07-05 22:16:31.459 +02:00 WARN context/context_channel.go:157 bad anonName(), do nothing
2019-07-05 22:16:31.460 +02:00 WARN context/context_channel.go:296 failed to get channel, type(test_dest)
resp: {{  0 false} {   } <nil>}, error: bad request module name(test_dest)
--- PASS: TestSendSync (5.00s)
PASS
ok  	github.com/kubeedge/beehive/pkg/core/context	6.022s
?   	github.com/kubeedge/beehive/pkg/core/model	[no test files]
?   	github.com/kubeedge/beehive/test/cmd	[no test files]
?   	github.com/kubeedge/beehive/test/modules	[no test files]
make: *** [Makefile:44: test] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,0 +1,63 @@
+#
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+PKG=github.com/kubeedge/beehive
+
+# Project packages.
+PACKAGES=$(shell go list ./... | grep -v "/vendor/")
+INTEGRATION_PACKAGE=${PKG}
+
+#Replaces ":" (*nix), ";" (windows) with newline for easy parsing
+GOPATHS=$(shell echo ${GOPATH} | tr ":" "\n" | tr ";" "\n")
+
+# Flags passed to `go test`
+TESTFLAGS ?= -v
+
+.PHONY: all test lint depcheck benchmark coverage
+
+.DEFAULT: default
+
+all: test lint depcheck ## default
+
+depcheck: ## Check if imports, Gopkg.toml, and Gopkg.lock are in sync
+	dep check
+
+lint: ## golangci-lint check
+	golangci-lint run --disable-all -E gofmt -E golint ./...
+	go vet ./...
+
+test: ## test case
+	@go test ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES})
+
+# https://deepzz.com/post/study-golang-test.html
+# https://deepzz.com/post/the-command-flag-of-go-test.html
+benchmark: ## run benchmarks tests
+	@go test ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES})  -bench . -run Benchmark 
+	
+coverage: ## generate coverprofiles from the unit tests, except tests that require root
+	@rm -f coverage.txt
+	@go test -i ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES}) 2> /dev/null
+	@( for pkg in $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES}); do \
+		go test ${TESTFLAGS} \
+			-cover \
+			-coverprofile=profile.out \
+			-covermode=atomic $$pkg || exit; \
+		if [ -f profile.out ]; then \
+			cat profile.out >> coverage.txt; \
+			$(RM) profile.out; \
+		fi; \
+	done )

--- a/pkg/common/config/config.go
+++ b/pkg/common/config/config.go
@@ -46,6 +46,10 @@ func InitializeConfig() {
 		envSource := envconfigsource.NewEnvConfigurationSource()
 		CONFIG.AddSource(envSource)
 		confLocation := getConfigDirectory() + "/conf"
+		_, err = os.Stat(confLocation)
+		if !os.IsExist(err) {
+			os.Mkdir(confLocation, os.ModePerm)
+		}
 		err = filepath.Walk(confLocation, func(location string, f os.FileInfo, err error) error {
 			if f == nil {
 				return err

--- a/pkg/common/config/config_test.go
+++ b/pkg/common/config/config_test.go
@@ -16,13 +16,8 @@ import (
 
 func TestInitializeConfig(t *testing.T) {
 	configDir := util.GetCurrentDirectory() + "/conf"
-	//create if not exists
-	_, err := os.Stat(configDir)
-	if !os.IsExist(err) {
-		os.Mkdir(configDir, os.ModePerm)
-	}
 	defer os.RemoveAll(configDir)
-	err = prepareConfigFile(configDir)
+	err := prepareConfigFile(configDir)
 	addSources(configDir)
 	if err != nil {
 		t.Error(err)

